elytron/WFLY-14984-encrypted-expression-for-client-config.adoc

```diff
@@ -0,0 +1,171 @@
+                <clear-password password="password"/>
+            </protection-parameter-credentials>
+        </credential-store>
+    </credential-stores>
+    <expression-resolvers default-resolver="expression-resolver-name">
+        <expression-resolver name= “expression-resolver-name” credential-store= “credential-store-name" alias=“key”/>
+        <expression-resolver name= “expression-resolver-name-2” credential-store= “credential-store-name" alias=“example”/>
+    </expression-resolvers>
+</encrypted-expressions>
+```
+** Note that the initial password for the credential-store used for the expression-resolver would need to be specified in clear-text if it is a keystoreCredentialStore. However, if it is of PropertiesCredentialStore type, then the password would no longer need to be specified. 
+
+** Once parsing is done, the expression-resolver would need to be created programmatically and be connected with the associated secret-key aliases from a credential-store. 
+
+* Once this has been configured, secret keys can be added to the credential store under different aliases and be used for client configuration as follows: 
+```
+<credential-store name="credential-store-name" type="JCEKS" provider="provider-name" >
+    <attributes>
+        <attribute name="location" value="path/to/credential-store-name.cs" />
+    </attributes>
+    <protection-parameter-credentials>
+	    <clear-password password="${ENC::expression-resolver-name:encrypted-expression}"" />
+    </protection-parameter-credentials>
+</credential-store>
+```
+
+* In cases where the default-resolver is to be used for the expression, the field for "expression-resolver-name" can be left empty. Instead, the format would be as follows: 
+```
+    <protection-parameter-credentials>
+	    <clear-password password="${ENC::encrypted-expression}"" />
+    </protection-parameter-credentials>
+```
+
+* The secret keys that would be used to encrypt expressions would be created using the `--generate-secret-key` command and can be read using the `--export-secret-key` command as follows using the WildFly-Elytron tool: 
+```
+$ bin/elytron-tool.sh credential-store --generate-secret-key example --location=standalone/configuration/mycredstore.cs
+```
+* In order to encrypt a clear-text expression, `--encrypt` option in the wildfly-elytron tool can be used as follows: 
+```
+$ bin/elytron-tool.sh credential-store --location standalone/configuration/store-one.cs --type PropertiesCredentialStore --encrypt key
+Clear text value:
+Confirm clear text value:
+Clear text encrypted to token 'RUxZAUMQvGzk6Vaadp2cahhZ6rlPhHOZcWyjXALlAthrENvRTvQ=' using alias 'key'.
+```
+
+* expression-resolvers should be backed by an entry from a credential-store of type KeyStoreCredentialStore. However, when using a KeyStoreCredentialStore type credential store, the protection parameter, which is the password, needs to be specified in clear-text inside the <encrypted-expressions> tag. 
```

Comment:
expression-resolvers can reference any type of credential-store right? Here, KeyStoreCredentialStore is mentioned specifically.

Reply:
Changed it to "However, when using a other types of credential stores that are protected by a protection parameter, such as a password, that needs to be specified in clear-text inside the <encrypted-expressions> tag. "